headersprovider/headers_provider.go

```diff
@@ -18,18 +16,12 @@ type grpcHeaderProvider struct {
 	headers map[string]string
 }
 
-func newGrpcHeaderProvider(headers map[string]string) *grpcHeaderProvider {
-	provider := &grpcHeaderProvider{headers}
-	provider.headers[h.CallerTypeHeaderName] = h.CallerTypeHeaderCLI
-	return provider
-}
-
 func (a grpcHeaderProvider) GetHeaders(ctx context.Context) (map[string]string, error) {
 	return a.headers, nil
 }
 
 func SetGRPCHeadersProvider(headers map[string]string) {
-	headersProvider = newGrpcHeaderProvider(headers)
+	headersProvider = &grpcHeaderProvider{headers}
```

Comment:
where do we inject that caller-type header?

Reply:
It's included with the other version and client-name headers: https://github.com/temporalio/cli/blob/main/headers/headers.go#L37